crates/trigger/src/cli/stdio.rs

```diff
@@ -95,11 +122,16 @@ impl<F: RuntimeFactors, U> ExecutorHooks<F, U> for StdioLoggingExecutorHooks {
         configured_app: &spin_factors::ConfiguredApp<F>,
     ) -> anyhow::Result<()> {
         self.validate_follows(configured_app.app())?;
+
         if let Some(dir) = &self.log_dir {
             // Ensure log dir exists if set
             std::fs::create_dir_all(dir)
                 .with_context(|| format!("Failed to create log dir {}", quoted_path(dir)))?;
 
+            if self.truncate_log {
+                self.truncate_log_files(&self.log_dir);
```

Comment:
You can pass `dir` here and save the redundant pattern match in `truncate_log_files`.